api/controllers/apps.go

```diff
@@ -47,11 +47,15 @@ func AppShow(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 func AppCreate(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 	name := r.FormValue("name")
 
-	app := &models.App{
-		Name: name,
+	// Early check for unbound app only.
+	app, err := models.GetAppFast(name, false, true)
```

Comment:
How about breaking up GetAppFast into smaller and more descriptive functions? Since this lookup is for backwards compatibility only, the call logic could make it more obvious:

```
if app, err := models.GetAppUnbound(); err != nil {
  return httperr.Errorf(403, "there is already a legacy app named %s (%s). We recommend you delete this app and create it again.", name, app.Status)
}

err := app.Create()
...
```